publication-model/src/main/java/no/unit/nva/model/associatedartifacts/file/File.java

```diff
@@ -418,5 +422,17 @@ public File buildUnpublishableFile() {
             return new AdministrativeAgreement(identifier, name, mimeType, size, license, administrativeAgreement,
                                                publisherVersion, embargoDate, uploadDetails);
         }
+
+        public File build(Class<? extends File> clazz) {
+            if (clazz.equals(PublishedFile.class)) {
+                return buildPublishedFile();
+            } else if (clazz.equals(UnpublishedFile.class)) {
+                return buildUnpublishedFile();
+            } else if (clazz.equals(AdministrativeAgreement.class)) {
+                return buildUnpublishableFile();
+            } else {
+                throw new IllegalArgumentException("Invalid file type");
+            }
+        }
```

Comment:
I think this builder is a more convenient to use when you consume file without knowing its actual type.